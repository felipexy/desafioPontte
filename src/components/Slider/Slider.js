import React, { useState } from 'react';
import './Slider.scss';
import { useDispatch } from 'react-redux';

const Slider = ({min, max}) => {

    let [rangeVal, setRangeVal] = useState(2);
    const dispatch = useDispatch();

    const handleChange = (event) => {
        setRangeVal(event.target.value)
        console.log(event.target.value);
        document.getElementById("Slider").style.background = `linear-gradient(90deg, rgb(139,185,38) ${event.target.value*25}%, rgb(118,0,117) ${event.target.value*25}%)`       
        dispatch({ type: 'CHANGE_ID', id: event.target.value } )
    }



    return (
        <div>
            <div className="sliderContainer"> 
                <input id="Slider" type="range" className="slider" min={min} max={max} value={rangeVal} step="1"
                onChange={(event) => handleChange(event)}/>
            </div>
            <div>
                <ul className="range-labels">
                    <li></li>
                    <li></li>
                    <li></li>
                    <li></li>
                    <li></li>
                </ul>
            </div>    
        </div>
    );
  };

  export default Slider;



  