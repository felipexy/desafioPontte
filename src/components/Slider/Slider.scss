.sliderContainer{
  -webkit-appearance: none;
  display: flex;
  flex-direction: column;
  background: linear-gradient(90deg, rgb(139,185,38) 50%, rgb(118,0,117) 50%);
  width: 92%;
  height: 4px;
  margin-right: auto;
  margin-left: auto;
}

.slider{
  -webkit-appearance: none;
  margin-right: auto;
  margin-left: auto;
  margin-top: 0px;
  width: 98%;
  height: 4px;
  background: linear-gradient(90deg, rgb(139,185,38) 50%, rgb(118,0,117) 50%);
  -webkit-transition: 0.2s;
  transition: opacity 0.2s;
  border-radius: 12px;
}

.slider::-webkit-slider-thumb {
  -webkit-appearance: none;
  width: 18px;
  height: 18px;
  border-radius: 50%;
  background: rgb(3, 3, 58);
  cursor: pointer;
  display: block;
}

.slider::-moz-range-thumb {
  -webkit-appearance: none;
  width: 18px;
  height: 18px;
  border-radius: 50%;
  background: rgb(3, 3, 58);
  cursor: pointer;
  display: block;
}

.flexLabels{
  display: flex;
  margin-top: -0.9%;
}

.pull-left{
  position: relative;
  margin-left: 4%;
  font-size: 1.2vw;
}

.pull-right{
  position: relative;
  margin-left: 77%;
  font-size: 1.2vw;
}

.range-tickers {
  display: flex;
  padding: 0;
  margin-left: -5.4%;
  margin-right: -5.4%;
  list-style: none;
  margin-bottom: -22px;
}

.ticker{
  position: relative;
  width: 100%;
  &::before {
    position: absolute;
    top: -20px;
    right: 0;
    left: 0;
    content: "";
    margin: 0 auto;
    width: 4px;
    height: 4px;
    background: white;
    border-radius: 50%;
  }
}

.tickerInvisible{
  position: relative;
  width: 100%;
  &::before {
    position: absolute;
    top: -20px;
    right: 0;
    left: 0;
    content: "";
    margin: 0 auto;
    width: 4px;
    height: 4px;
    background: white;
    border-radius: 50%;
  }
  z-index: -1;
}