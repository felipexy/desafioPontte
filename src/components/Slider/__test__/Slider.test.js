import React from 'react';
import ReactDOM from 'react-dom';
import Slider from '../Slider';
import { render, cleanup, waitFor, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import wrapper from 'enzyme';

afterEach(cleanup);

const apiData = `{
    "prazos": [
        60,
        90,
        120,
        150,
        180
    ],
    "valoresEmprestimo": [
        70000,
        85000,
        100000,
        115000,
        130000
    ],
    "valoresEmprestimeBruto": [
        78919.63,
        95114.83,
        110841.81,
        126366.55,
        141891.29  
    ],
    "parcelas": [
        [
            2287.12,
            2736.97,
            3173.81,
            3605.04,
            4036.26
        ],
        [
            1848.68,
            2208.55,
            2558.02,
            2903,
            3247.98
        ],
        [
            1629.46,
            1944.35,
            2250.13,
            2551.98,
            2853.83 
        ],
        [
            1497.93,
            1785.82,
            2065.39,
            2341.37,
            2617.35
        ],
        [
            1410.24,
            1680.14,
            1942.24,
            2200.96,
            2459.69
        ]
    ]
}`;

const startingState = { id: 2, valid: false, data: apiData, loading: false };

function reducer(state = startingState, action) {
  switch (action.type) {
    case 'CHANGE_ID':
      return {
        id: action.id,
        valid: state.valid,
        data: state.data,
        loading: state.loading,
      };
    case 'CHANGE_VALID':
      return {
        id: state.id,
        valid: action.valid,
        data: state.data,
        loading: state.loading,
      };
    case 'LOAD_DATA':
      return {
        id: state.id,
        valid: state.valid,
        data: action.data,
        loading: action.loading,
      };
    default:
      return state;
  }
}

function renderWithRedux(
  component,
  { initialState, store = createStore(reducer, initialState) } = {}
) {
  return {
    ...render(<Provider store={store}>{component}</Provider>),
  };
}

describe('Tests for CardMonths component with Redux', () => {
  it('Renders with Redux', () => {
    const { getByTestId } = renderWithRedux(<Slider />);
  });

  it('Check if onChange is called', async () => {
    const { getByTestId } = renderWithRedux(
      <Slider
        min="0"
        maxPrazos="4"
        minValue="70.000,00"
        maxValue="130.000,00"
      />
    );

    const sldTst = await waitFor(() => getByTestId('sliderTest'));
    fireEvent.change(sldTst, { target: { value: '5' } });
    // expect(sldTst).toBeTruthy();
  });

  it('Matches snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={createStore(reducer, startingState)}>
          <Slider />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
