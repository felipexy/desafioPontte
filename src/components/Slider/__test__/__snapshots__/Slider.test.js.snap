// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for CardMonths component with Redux Matches snapshot 1`] = `
<div>
  <div
    className="sliderContainer"
  >
    <input
      className="sliderContainer__slider"
      data-testid="sliderTest"
      id="Slider"
      onChange={[Function]}
      step="1"
      style={
        Object {
          "background": "linear-gradient(90deg, rgb(139,185,38) NaN%, rgb(118,0,117) NaN%)",
        }
      }
      type="range"
      value={NaN}
    />
  </div>
  <ul
    className="tickList"
  />
  <div
    className="flexLabels"
  >
    <p
      className="flexLabels__pull-left"
    >
      R$
    </p>
    <p
      className="flexLabels__pull-right"
    >
      R$
    </p>
  </div>
</div>
`;
